### PR TITLE
[ci] Separate MacOS flaky tests into a new test suite

### PR DESCRIPTION
**Pull request by khluu**
## Why are these changes needed?
- These flaky tests cause the whole test suite to soft fail.
- It's better to separate these flaky tests into another test suite so we can go back to enforce hard fail on other test suites.

## Changes
- A script to automatically query current flaky tests.
- Exclude flaky tests (from the query) from the macos test suites.
- Add a new test suite to contain only flaky tests which runs separately from other macos test suites.

## Checks
- `macos_ci.sh` was able to query flaky tests just fine.
- Flaky test suite contains expected tests.
- Other test suites exclude expected flaky tests.

### CODE REVIEW

File: ci/ray_ci/automation/query_flaky_macos_test.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    Input:
+    tests: List of Test objects
+    prefix: Prefix of the test names
+    """
+    test_names = [test.get_name().replace(prefix, '') for test in tests]
+    return json.dumps(test_names)
+
+if __name__ == "__main__":
+    main()
```
Comment (can-anyscale) [inline ci/ray_ci/automation/query_flaky_macos_test.py:56]: nit: add new line at the end

Comment (khluu) [reply ci/ray_ci/automation/query_flaky_macos_test.py]: done!

**Comment by aslonnie** [comment]
please fix code format lint error.

If needs running `scripts/format.sh`

File: ci/ray_ci/automation/query_flaky_macos_test.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+def main() -> None:
+    logger.info("Start querying flaky tests ...")
+    init_global_config(bazel_runfile("release/ray_release/configs/oss_config.yaml"))
+    tests = query_all_tests()
+    flaky_tests = filter_flaky_tests(tests)
+    logger.info("Number of flaky tests found: " + str(len(flaky_tests)))
+    logger.info("List of flaky tests: ")
+    print(retrieve_test_names(flaky_tests, 'darwin://'))
+
+def query_all_tests() -> List[Test]:
```
Comment (aslonnie) [inline ci/ray_ci/automation/query_flaky_macos_test.py:20]: move these functions into `ray_release.test` maybe ? or a `test_utils` file somewhere, make it a library, and add unit tests?

**Comment by khluu** [comment]
@aslonnie  I just reformatted the code and moved the test query function over to `ray_release.test` as part of `Test` class. The new test suite is taking really long and failing to run on `premerge` (though it works locally) so I'll look into that.

**Review by aslonnie — COMMENTED** [review]
talked offline, let's do this in 2 steps:

1. split the `bazel test` commands into two steps: `bazel query` + `bazel test`.
2. insert a filtering step between the two steps.

**Pull request closed without merging**